zcash_client_backend/src/zip321.rs

```diff
@@ -113,43 +112,67 @@ impl Memo {
     pub fn from_base64(s: &str) -> Result<Self, MemoError> {
         base64::decode_config(s, base64::URL_SAFE_NO_PAD)
             .map_err(MemoError::InvalidBase64)
-            .and_then(Memo::from_vec)
+            .and_then(|b| RawMemo::from_bytes(&b))
     }
 }
 
-impl Debug for Memo {
+impl Debug for RawMemo {
     fn fmt(&self, f: &mut fmt::Formatter) -> Result<(), fmt::Error> {
-        f.debug_struct("Memo")
+        f.debug_struct("RawMemo")
             .field("memo", &format!("{:?}...", &self.0[0..17]))
             .finish()
     }
 }
 
-impl PartialEq for Memo {
+impl PartialEq for RawMemo {
     fn eq(&self, other: &Self) -> bool {
         self.to_base64() == other.to_base64()
     }
 }
 
-impl Eq for Memo {}
+impl FromStr for RawMemo {
+    type Err = MemoError;
+
+    fn from_str(memo: &str) -> Result<Self, Self::Err> {
+        RawMemo::from_str(memo)
+    }
+}
+
+impl Eq for RawMemo {}
 
-impl PartialOrd for Memo {
+impl PartialOrd for RawMemo {
     fn partial_cmp(&self, other: &Self) -> Option<Ordering> {
         Some(self.to_base64().cmp(&other.to_base64()))
     }
 }
 
-impl Ord for Memo {
+impl Ord for RawMemo {
     fn cmp(&self, other: &Self) -> Ordering {
         self.partial_cmp(other).unwrap()
     }
 }
 
+// RawMemo is somewhat duplicative of the `Memo` type
+// in crate::note_encryption but as that's actively being
+// updated at time of this writing, these functions provide
+// shims to ease future use of those
+pub fn memo_from_vec(v: &[u8]) -> Result<RawMemo, MemoError> {
+    RawMemo::from_bytes(v)
+}
+
+pub fn memo_to_base64(memo: &RawMemo) -> String {
+    memo.to_base64()
+}
+
+pub fn memo_from_base64(s: &str) -> Result<RawMemo, MemoError> {
+    RawMemo::from_base64(s)
+}
+
 #[derive(Debug, PartialEq)]
 pub struct Zip321Payment {
     recipient_address: Address,
     amount: Amount,
-    memo: Option<Memo>,
+    memo: Option<RawMemo>,
```

Comment:
Is there a strong consensus that this field should have the identifier `memo`?

Reply:
I think so.. do you have another suggestion?